### PR TITLE
SophosCentral: Update object IDs to names in alerts commands

### PR DESCRIPTION
**Pull request by yaakovi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
added the name of the person/managed agent (device) of the alerts.
used the integration context as cache.
tested in a customer's environment.

will update the README with the new outputs after the docs-review.

### CODE REVIEW

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py [hunk 1]
```diff
@@ -76,6 +77,30 @@ def get_jwt_token_from_api(client_id: str, client_secret: str) -> dict:
             raise DemistoException(response.text)
         return response.json()
 
+    def convert_id_to_name(self, object_name: str, object_id: str, url_suffix: str, field_key: str) -> str:
```
Comment (teizenman) [inline Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:80]: I think that docstring will be helpful here.

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py [hunk 2]
```diff
@@ -76,6 +77,30 @@ def get_jwt_token_from_api(client_id: str, client_secret: str) -> dict:
...
+
+            demisto.debug(f'did not find object id in {object_name} cache, retreiving from API')
+            response = self._http_request(
+                method='GET', url_suffix=url_suffix,
+                headers=self.headers,
+            )
+            object_mapping[object_id] = response.get(field_key, '')
+            return object_mapping[object_id]
+        except Exception as exc:
+            demisto.debug(f'failed to convert the {object_name} id, Error: {exc}\nTrace{traceback.format_exc()}')
```
Comment (teizenman) [inline Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:94]:
```suggestion
            demisto.debug(f'failed to convert the {object_name} id, Error: {exc}\n{traceback.format_exc()}')

```

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py [hunk 3]
```diff
@@ -637,17 +679,27 @@ def create_alert_output(item: Dict, table_headers: List[str]) -> Dict[str, Optio
         object_data (dict(str)): The output dictionary.
     """
     alert_data = {field: item.get(field) for field in table_headers + ['groupKey', 'product']}
+
     managed_agent = item.get('managedAgent')
     if managed_agent:
+        managed_agent_morphed_id = flip_chars(managed_agent.get('id', ''))
         alert_data['managedAgentId'] = managed_agent.get('id')
+        alert_data['managedAgentIdMorphed'] = managed_agent_morphed_id
```
Comment (teizenman) [inline Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:687]: Do you want it to be an empty string while the regular ID can be `None`?

File: Packs/SophosCentral/ReleaseNotes/1_0_5.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+##### Sophos Central
+- Added the *managedAgentName* and the *personName* to the following commands:
```
Comment (teizenman) [inline Packs/SophosCentral/ReleaseNotes/1_0_5.md:4]: What about the `Morphed`?

Comment (yaakovi) [reply Packs/SophosCentral/ReleaseNotes/1_0_5.md:4]: not declaring them as they are a workaround and shouldn't be used.

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py (hunk 3 above)
Comment (yaakovi) [reply Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:687]: I think this is fine, as the id will always exist as long as the object is returned from the API.

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py [hunk 5]
```diff
@@ -1,1811 +1,1839 @@
...
+        object_data (dict(str)): The output dictionary.
+    """
+    alert_data = {field: item.get(field) for field in table_headers + ['groupKey', 'product']}
+
+    managed_agent = item.get('managedAgent')
+    if managed_agent:
+        managed_agent_morphed_id = flip_chars(managed_agent.get('id', ''))
+        alert_data['managedAgentId'] = managed_agent.get('id')
+        alert_data['managedAgentIdMorphed'] = managed_agent_morphed_id
+        alert_data['managedAgentName'] = client.get_managed_agent_name(managed_agent.get('id'))
```
Comment (reutshal) [inline Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:703]:
```suggestion
        alert_data['managedAgentName'] = client.get_managed_agent_name(managed_agent.get('id', ''))
```

**Comment by ShirleyDenkberg** [comment]
@reutshal @teizenman Doc review completed.

File: Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py [hunk 6]
```diff
@@ -1,1811 +1,1839 @@
...
+        alert_data['tenantId'] = tenant.get('id')
+        alert_data['tenantIdMorphed'] = flip_chars(tenant.get('id', ''))
+        alert_data['tenantName'] = tenant.get('name')
+
+    person = item.get('person')
+    if person:
+        person_morphed_id = flip_chars(person.get('id', ''))
+        alert_data['person'] = person.get('id')
+        alert_data['personIdMorphed'] = person_morphed_id
+        alert_data['personName'] = client.get_person_name(person.get('id'))
```
Comment (reutshal) [inline Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py:717]:
```suggestion
        alert_data['personName'] = client.get_person_name(person.get('id', ''))
```

**Review by teizenman — APPROVED** [review]

File: Packs/SophosCentral/ReleaseNotes/1_0_5.md (hunk 4 above)
Comment (reutshal) [reply Packs/SophosCentral/ReleaseNotes/1_0_5.md:4]:
```suggestion
- Added the *managedAgentName* and the *personName* output paths to the following commands:
```

**Review by reutshal — APPROVED** [review]

**Pull request merged**